File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = prices[0]
        max_prof = 0
        
        for price in prices:
            difference = price - buy
            if price < buy:
                buy = price
            elif difference > max_prof:
                max_prof = difference
        
        return max_prof

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
 let buy = prices[0]; //7
 let profit = 0;
 
 for (let price of prices) {
     let difference = price - buy;
     if (price < buy) {
         buy = price;
     } else if (difference > profit) {
         profit = difference;
     }
 }

 return profit;
    
};